treeherder/perfalert/perfalert/__init__.py

```diff
@@ -55,104 +72,107 @@ def calc_t(w1, w2, weight_fn=None):
     return delta_s / (((s1['variance'] / s1['n']) + (s2['variance'] / s2['n'])) ** 0.5)
 
 
-class Datum(object):
-    def __init__(self, push_timestamp, value, testrun_timestamp=None,
-                 testrun_id=None, revision_id=None, state='good'):
+class RevisionDatum(object):
+    '''
+    This class represents a specific revision and the set of values for it
+    '''
+    def __init__(self, push_timestamp, push_id, values):
+
         # Date code was pushed
         self.push_timestamp = push_timestamp
-        # Value of this point
-        self.value = value
 
-        # Which test run was this
-        self.testrun_id = testrun_id
-        # What revision this data is for
-        self.revision_id = revision_id
+        # What revision this data is for (usually, but not guaranteed
+        # to be increasing with push_timestamp)
+        self.push_id = push_id
+
+        # data values associated with this revision
+        self.values = copy.copy(values)
 
         # t-test score
         self.t = 0
+
         # Whether a perf regression is found
-        self.state = state
+        self.state = 'good'
```

Comment:
Silly legacy from old perf regression code. I'll just change this to a boolean called `change_detected`, which should be much easier to understand.